web3/_utils/rpc_abi.py

```diff
@@ -218,8 +218,9 @@ def apply_abi_formatters_to_dict(
         [abi_dict[field] for field in fields],
         [data[field] for field in fields],
     )
-    formatted_dict = dict(zip(fields, formatted_values))
-    return dict(data, **formatted_dict)
+    formatted_dict = data.copy()
+    formatted_dict.update(zip(fields, formatted_values))
```

Comment:
Instead of fully initializing a `formatted_dict` and then initializing a second new dict with it, we can just init one copy of the original dict and update it from a zip